core/drlibc/drlibc_xarch.asm

```diff
@@ -37,15 +37,15 @@
 #include "../arch/asm_defines.asm"
 START_FILE
 
-/* For AArch64, drlibc has no references to unexpected_return, and in fact we
- * have a relocation reachability error if we include it here (i#4304), so
- * we limit its use in drlibc to x86 or arm.
- */
-#ifndef AARCH64
 DECL_EXTERN(d_r_internal_error)
 
 /* For debugging: report an error if the function called by call_switch_stack()
  * unexpectedly returns.  Also used elsewhere.
+ * i#4304: When calling this function do not use short branch instructions,
+ * eg, conditional branch instructions on aarch64. They don't have enough
+ * bits of offset to reach this. Instead, use unconditional branch or
+ * call instructions (eg, b, bl on aarch64) that have a sufficient number of
```

Comment:
Better to not recommend a call instruction on x86 as it requires the stack and this routine is called in fragile places.  Consider call_switch_stack: if something goes wrong the stack may well be bogus and we'll crash if we execute a CALL.

Reply:
Done